### PR TITLE
fix for slack callback breaks in 2.4

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
fixes #30838

also restores some code incorrectly 'elsed' out of functioning path

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

callback/slack
##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

File: lib/ansible/plugins/callback/slack.py [hunk 1]
```diff
@@ -90,6 +82,17 @@ def __init__(self, display=None):
...
+        # This is a 6 character identifier provided with each message
+        # This makes it easier to correlate messages when there are more
+        # than 1 simultaneous playbooks running
+        self.guid = uuid.uuid4().hex[:6]
+
+    def set_options(self, options):
+
+        super(CallbackModule, self).set_options(options)
+
         self.webhook_url = self._plugin_options['webook_url']
```
Comment (sivel) [inline lib/ansible/plugins/callback/slack.py]: `webook_url` is still wrong here, should be `webhook_url`

Comment (bcoca) [reply lib/ansible/plugins/callback/slack.py]: dammit ... i forgot to fix the thing that triggered it all!

**Comment by hiciu** [comment]
Thanks! I can confirm that your version works ("works" as in does not throw an exception - I didn't test it further yet).

**Comment by bcoca** [comment]
backported to 2.4, will be in 2.4.1

**Pull request merged**